securemailbox/views/retrieve.py

```diff
@@ -1,11 +1,69 @@
-from flask import Blueprint, jsonify
+from flask import Blueprint, jsonify, request, json
+
+from ..models import Mailbox
+from ..models import Message
 
 from securemailbox import db
 
 # Create the `retrieve` blueprint
 retrieve_blueprint = Blueprint("retrieve", __name__)
 
-
 @retrieve_blueprint.route("/retrieve/", methods=["POST"])
 def retrieve():
-    return jsonify({"success": True, "error": None})
+
+    data = request.get_json(force=True, silent=True)
+
+    if data is None:
+        return jsonify({'success': False, 'error': 'Request must be valid json', 'data': None}), 400
+
+    # Check if fingerprint is present in json request body
+    if 'fingerprint' not in data:
+        return jsonify({'success': False, 'error': 'fingerprint not in request', 'data': None}), 400
+
+    # Check if fingerprint value is string variable before query
+    if(isinstance(data['fingerprint'], str) == False):
+        return jsonify({'success': False, 'error': 'fingerprint value not valid string', 'data': None}), 400
+
+    # search for fingerprint in mailbox table, assign to variable
+    mailbox_num = Mailbox.query.filter_by(fingerprint=data['fingerprint']).first()
```

Comment:
So, I _think_ you can do:
```suggestion
    mailbox_id = Mailbox.query(Mailbox.id).filter_by(fingerprint=data['fingerprint']).first()
```
Which will only select the id column of the row (note: I haven't tested this and it's based off [this](https://stackoverflow.com/a/37134248/4668680) SO answer).

I say this because it looks like you only using the mailbox id.